parser_library/src/processing/statement_analyzers/hit_count_analyzer.cpp

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) 2023 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+#include "hit_count_analyzer.h"
+
+#include <cassert>
+#include <iterator>
+#include <variant>
+
+#include "context/hlasm_context.h"
+#include "processing/statement.h"
+
+namespace hlasm_plugin::parser_library::processing {
+
+hit_count_analyzer::hit_count_analyzer(context::hlasm_context& ctx)
+    : m_ctx(ctx)
+{}
+
+namespace {
+constexpr auto get_core_stmt =
+    [](const context::hlasm_statement* stmt) -> std::optional<const semantics::core_statement*> {
```

Comment:
This could be just a simple const semantics::core_statement* return type, with nullptr being the false-like answer, right?